design_docs/generalized_horizontal_interpolation.md

```diff
@@ -0,0 +1,99 @@
+
+<h2>Design solution: Higher-order interpolation <br>
+Date last modified: 2017/02/25 <br>
+Contributors: Xylar Asay-Davis</h2>
+
+The approach will be to create SCRIP files (or, in the future for greater flexibility perhaps ESMF grid/mesh files) for the source and destination grids, then to use `ESMF_RegridWeightGen` to generate a mapping file.  `ESMF_RegridWeightGen` supports 5 interpolation methods---bilinear, patch, nearestdtos, neareststod, and conserve---and we would likely support at least bilinear, neareststod and conserve, and perhaps all 5.  The destination grid will be specified either by reading values from `lat` and `lon` coordinates of a NetCDF file or through config file options `lat` and `lon` that are typically expressions involving `numpy.arange` or `numpy.linspace`.
+
+Then, `ncremap` will be used to remap the desired list of variables from an MPAS NetCDF file to the desired destination grid.
+
+<h2>Design solution: Interpolation should handle periodic boundaries <br>
+Date last modified: 2017/02/25 <br>
+Contributors: Xylar Asay-Davis</h2>
+
+For now, periodic boundaries (except for the obvious one at +/- 180 longitude) will not be supported.  It appears that ESMF grid files do include support for periodic boundaries so the current solution should be relatively easy to extend to periodic boundaries in the future.
+
+<h2>Design solution: Interpolation should handle Cartesian meshes <br>
+Date last modified:  2017/02/25 <br>
+Contributors: Xylar Asay-Davis</h2>
+
+ESMF unstructured mesh files seem to support Cartesian coordinates.  This will be investigated if and when MPAS-Analysis can accommodate a test case with Cartesian coordinates.
+
+<h2>Design solution: Support for arbitrary output interpolation points <br>
+Date last modified: 2017/02/25 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+I do not intend to address this consideration in this project.  It may be that `ESMF_RegridWeightGen` can also be used to perform interpolation to arbitrary points (in particular, a set of points that are not cell centers or vertices of a mesh), but this is not yet clear to me.  If not, an alternative solution for arbitrary destination points will be needed.
+
+<h2>Design solution: Support caching results from any costly, one-time geometric computations <br>
+Date last modified: 2017/02/25 <br>
+Contributors: Xylar Asay-Davis</h2>
+
+This should be relatively easy to accommodate with `ESMF_RegridWeightGen` and `ncremap`.  The default behavior of the function for generating interpolation weights will be to do nothing if the mapping file already exists.  Further, we can support an optional config option that will point to an existing mapping file if one has already been generated and cached somewhere (e.g. in a shared directory).  Eventually, we will probably want to systematically store these mapping files for typical MPAS meshes and typical output grids, particularly for those that are expensive to generate.
```

Comment:
@xylar: this looks good to me. I only have one comment here: wouldn't you want to add a first check here on  whether the interpolated product (out of ncremap) already exists? If not, then go on check whether the mapping file exists?